acquia/cli
# CLI-1402: [push:db] Warn if view database connection details permission is missing

**Motivation**

Fixes CLI-1402

**Proposed changes**

- Update SDK to not throw a type error if connection details are missing so we can fail gracefully: https://github.com/typhonius/acquia-php-sdk-v2/pull/477
- Specifically check if details are missing and throw an exception if so (along with help text)
- Add a `RequireRemoteDb` attribute to warn users (via command help) that these perms are required

**Testing steps**


1. Follow the [contribution guide](https://github.com/acquia/cli/blob/HEAD/CONTRIBUTING.md#building-and-testing) to set up your development environment or [download a pre-built acli.phar](https://github.com/acquia/cli/blob/HEAD/CONTRIBUTING.md#automatic-dev-builds) for this PR.
2. Clear the kernel cache to pick up new and changed commands: `./bin/acli ckc`
3. Create a Cloud API user lacking the permission to view connection details.
4. Run `acli pull:db` (optionally include `--no-import` so you don't need a local DB), see that it runs successfully.
5. Run `acli push:db`, see that you get a graceful exception with details about the permission required.
6. Give yourself the permission.
7. Run `acli push:db` again and see that it runs successfully.
